vignettes/customise_query.Rmd

```diff
@@ -0,0 +1,409 @@
+#> # A tibble: 1 x 4
+#>   placename   oc_lat oc_lng oc_formatted                                       
+#>   <chr>        <dbl>  <dbl> <chr>                                              
+#> 1 Wall Street   40.7  -74.0 Wall St, Manhattan Community Board 1, NY 10005, USA
+```
+
+See [this blog post](https://blog.opencagedata.com/post/160294347883/shrtr-pls) for more information.
+
+## Vectorised arguments
+
+All function arguments mentioned above are vectorised, so you can send queries like this:
+
+
+```r
+oc_forward_df(
+  placename = c("New York", "Rio", "Tokyo"),
+  language = c("es", "de", "fr")
+)
+#> # A tibble: 3 x 4
+#>   placename oc_lat oc_lng oc_formatted                                                     
+#>   <chr>      <dbl>  <dbl> <chr>                                                            
+#> 1 New York    40.7  -74.0 Nueva York, Estados Unidos                                       
+#> 2 Rio        -22.9  -43.2 Rio de Janeiro, Região Metropolitana do Rio de Janeiro, Brasilien
+#> 3 Tokyo       35.7  140.  Tokyo, Japon
+```
+
+Or geocode placenames in a data frame:
+
+
+```r
+for_df <- data.frame(
+  location = c("Golden Gate Bridge", "Buckingham Palace", "Eiffel Tower"),
+  ccode = c("at", "cg", "be"))
+
+oc_forward_df(for_df, placename = location, countrycode = ccode)
+#> # A tibble: 3 x 5
+#>   location          ccode oc_lat oc_lng oc_formatted                                                                        
+#>   <chr>             <chr>  <dbl>  <dbl> <chr>                                                                               
+#> 1 Golden Gate Brid~ at     48.0   15.6  Karer, Golden Gate Bridge, 3180 Gemeinde Lilienfeld, Austria                        
+#> 2 Buckingham Palace cg     -4.80  11.8  Buckingham Palace, Boulevard du Général Charles de Gaulle, Pointe-Noire, Republic o~
+#> 3 Eiffel Tower      be     50.9    4.34 Eiffel Tower, Avenue de Bouchout - Boechoutlaan, Brussels, Belgium
+```
+
+That also works with `oc_reverse_df()`, of course. 
+
+
+```r
+rev_df <- data.frame(
+  lat = c(51.952659, 41.401372, 0),
+  lon = c(7.632473, 2.128685, 0))
+
+oc_reverse_df(rev_df, lat, lon, language = "native")
+#> # A tibble: 3 x 3
+#>     lat   lon oc_formatted                                        
+#>   <dbl> <dbl> <chr>                                               
+#> 1  52.0  7.63 Friedrich-Ebert-Straße 7, 48153 Münster, Deutschland
+#> 2  41.4  2.13 Carrer de Calatrava, 68, 08017 Barcelona, España    
+#> 3   0    0    Null Island
+```
+
+## Further information
+
+For further information about the output and the query parameters, see the [API docs](https://opencagedata.com/api) and the [OpenCage FAQ](https://opencagedata.com/faq). When building queries, OpenCage's [best practices](https://opencagedata.com/api#bestpractices) can be very useful, as well as their guide to [geocoding accuracy](https://opencagedata.com/guides/how-to-think-about-geocoding-accuracy).
```

Comment:
btw is there any user forum or so?

Reply:
I don't think there is anything besides https://opencagedata.com/contact and that seems to be [intentional](https://github.com/OpenCageData/opencagedata-misc-docs/issues/7#issuecomment-700535181).